### PR TITLE
Added a Relocation and skytower side mission 

### PR DESCRIPTION
**Pull request by Gkramme**
This is a colonist-type job that has you relocate people from overcrowded planets. Please excuse the Spartan commit, I'm stil learning this.

### CODE REVIEW

**Comment by endless-sky** [comment]
Maybe call them something other than "displaced"? That implies they've been driven out by war or natural disaster.

**Comment by Gkramme** [comment]
I wouldn't know what to call them, unless homeless of jobless works.

**Comment by Isaacssv552** [comment]
Jobless seems like the best bet, there's already one mission (not a job, one-time) about moving people to planets with better work.

**Comment by Gkramme** [comment]
@endless-sky is this ready for merge?

File: data/transport missions.txt [hunk 1]
```diff
@@ -1246,3 +1246,70 @@ mission "Terraforming 5"
...
+			choice
+				`	"Yes, indeed."`
+				`	"Is this a skytower?"`
+				
+			`	"It is," he says. "Now, I know what you're thinking. 'Wern't skytowers prohibited when the one in 2235 fell?' They were. But, with the ever growing population, this planet is turning into a city world. On top of that, the emmisions from the billions of people on this world are destroying the atmosphere, and the oceans are rising steadily. We need to get lots of people off the planet, and fast too if we want to save our homeworld from becoming a ecological disaster."`
+			choice
+				`	"This is scary stuff. Is there any way I can help?"`
+                                `       "Wow. What can we do?"`
+
+                       `        "There is somthing you can do. We need some cargo ships escorted to Alphard to pick up some super-strong materials we will use to build this." He gestures toward the tower. "I also need you to bring some scientists there as well. Are you up to the challange?"`
```
Comment (jafdy) [inline data/transport missions.txt]: That is impressively unscientific. Try `high-strength alloy`.

**Comment by endless-sky** [comment]
Only one of these missions will actually be offered. The others have various problems with them: planet names that aren't capitalized, syntax errors, improper indentation, grammatical mistakes, misspellings, etc. It might be better to create this as a plugin and have some people in the forums test it and work out the issues before proposing it as a pull request.

File: data/transport missions.txt [hunk 2]
```diff
@@ -1267,12 +1267,12 @@ mission "Terraforming 5"
...
 			choice
 				`	"This is scary stuff. Is there any way I can help?"`
                                 `       "Wow. What can we do?"`
 
-                       `        "There is somthing you can do. We need some cargo ships escorted to Alphard to pick up some super-strong materials we will use to build this." He gestures toward the tower. "I also need you to bring some scientists there as well. Are you up to the challange?"`
+                       `        "There is somthing you can do. We need some cargo ships escorted to Alphard to pick up some super-strong nanotube buckyballs that hopefully should help strengthen this." He gestures toward the tower. "I also need you to bring some scientists back as well. Are you up to the challange?"`
```
Comment (jafdy) [inline data/transport missions.txt]: A buckyball is spherical and a nanotube is a tube. A material can't be both.

File: data/transport missions.txt [hunk 3]
```diff
@@ -1295,12 +1295,14 @@ mission "Building Skytower 2"
 	to offer
 		has "building skytower 1: done"
 	passengers 4
+       
+        on complete
         set licence: navy
 	
 	on offer
-		conversation
+		diolouge
```
Comment (jafdy) [inline data/transport missions.txt]: dialog not diolouge

File: data/transport missions.txt [hunk 4]
```diff
@@ -1246,3 +1246,72 @@ mission "Terraforming 5"
...
+ "Building Skytower 1"
+	name "Building Skytower [1]"
+	description "Escort a convoy of freighters to <destination>, where they will pick up materials for an new skytower on Earth."
+	minor
+	source earth
+	destination alphard
+	to offer
+		random < 10
+                reputation republic: 1<
+                has "event: war ends: done"
```
Comment (jafdy) [inline data/transport missions.txt]: 1258 and 1259. I think you have a space in there.

File: data/transport missions.txt [hunk 5]
```diff
@@ -1246,3 +1246,72 @@ mission "Terraforming 5"
...
+				`	"Is this a skytower?"`
+				
+			`	"It is," he says. "Now, I know what you're thinking. 'Wern't skytowers prohibited when the one in 2235 fell?' They were. But, with the ever growing population, this planet is turning into a city world. On top of that, the greenhouse gas emmisions from the billions of people on this world are destroying the atmosphere, and the oceans are rising steadily. We need to get lots of people off the planet, and fast too if we want to save our homeworld from becoming a ecological disaster."`
+			choice
+				`	"This is scary stuff. Is there any way I can help?"`
+                                `       "Wow. What can we do?"`
+
+                       `        "There is somthing you can do. We need some cargo ships escorted to Alphard to pick up some super-strong nanotube buckyballs that hopefully should help strengthen this." He gestures toward the tower. "I also need you to bring some scientists back as well. Are you up to the challange?"`
+                        choice
+                                `       "Yes I am"`
```
Comment (jafdy) [inline data/transport missions.txt]: 1273 and 1277. More spaces I think.

File: data/transport missions.txt [hunk 6]
```diff
@@ -1246,3 +1246,72 @@ mission "Terraforming 5"
...
+
+
+mission "Building Skytower 2"
+	landing
+	name "building skytower [2]"
+	description "Transport the team of scientists home to <destination>."
+	source
+		alphard
+	destination
+		earth
```
Comment (jafdy) [inline data/transport missions.txt]: Alphard and Earth. Need to be on the same line as source and destination. Planet names need to start with capital letters.

File: data/transport missions.txt [hunk 7]
```diff
@@ -1246,3 +1246,72 @@ mission "Terraforming 5"
...
+	to offer
+		has "building skytower 1: done"
+	passengers 4
+       
+        on complete
+        set licence: navy
```
Comment (jafdy) [inline data/transport missions.txt]: Extra spaces here as well.

**Comment by Gkramme** [comment]
Yea. I made a lot of mistakes :/ I'm new to this.

File: data/transport missions.txt (hunk 5 above)
Comment (Gkramme) [inline data/transport missions.txt]: I can't see it when I edit it.

Comment (jafdy) [inline data/transport missions.txt]: They are indented by slightly different amounts. Something must be there.

Comment (lehgogh) [inline data/transport missions.txt]: Yep; each of those two lines has one extra space.

File: data/transport missions.txt [hunk 8]
```diff
@@ -1246,3 +1246,71 @@ mission "Terraforming 5"
...
+
+
+mission "Building Skytower 2"
+	landing
+	name "building skytower [2]"
+	description "Transport the team of scientists home to <destination>."
+	source Alphard
+	destination Earth
+	to offer 
+                has "building skytower 1: done"
```
Comment (lehgogh) [inline data/transport missions.txt]: One too many spaces on this line, as well as lines 1298 and 1299.

File: data/transport missions.txt [hunk 9]
```diff
@@ -1246,3 +1246,71 @@ mission "Terraforming 5"
...
+                        choice
+                                `       "Yes I am"`
+				`       "Sorry, but I have somthing urgent to do"` 
+                                decline	
+			`       "Thanks!" He says, "By the way, I'm admrial Michel Farford." "<first> <last>." you say. You shake hands, and you show Michel to his bunk."`
+	
+	on complete
+		payment 200000
+ 
+               #Fleet of freighter ships is needed here before merge
```
Comment (jafdy) [inline data/transport missions.txt]: Try 3-5 Bulk freighters. You may want to put in some Pirate fleets in both missions to cause trouble. You will also need a `to fail` condition.

File: data/transport missions.txt [hunk 10]
```diff
@@ -1246,3 +1246,71 @@ mission "Terraforming 5"
...
+	
+	on offer
+		diolouge
+			`As you are landing, you can see many Republic ships lofting materials to and from the surface. This project has a huge imprint on this and many other planets. `
+			`You pick up the scientests and watch the cargo ships pick up the nanotubes. You prepare for the flight back to Earth.`
+				accept
+	
+	on complete
+		payment 400000
+		conversation `Michel thanks you for escorting the ships, and gratefully hands you <payment>. Before you walk away, he clasps your shoulder and says: "You know, we're very grateful for the help you gave us, so I think we can give you a little somthing other than just a payment." He pulls out a slip of paper and writes on it. "<firt> <last>, right?" "Yes," you say. "Here," He hands you an Navy licence, "It won't buy you a cruiser, but it's the most I can give you."`
```
Comment (jafdy) [inline data/transport missions.txt]: Normally when a different person talks you put it on a new line.

File: data/transport missions.txt [hunk 11]
```diff
@@ -1246,3 +1246,71 @@ mission "Terraforming 5"
...
+			choice
+				`	"Yes, indeed."`
+				`	"Is this a skytower?"`
+				
+			`	"It is," he says. "Now, I know what you're thinking. 'Wern't skytowers prohibited when the one in 2235 fell?' They were. But, with the ever growing population, this planet is turning into an urban world. On top of that, the greenhouse gas emmisions from the billions of people that call this world home are destroying the atmosphere, and the oceans are rising steadily. We need to get lots of people off the planet, and fast too if we want to save our homeworld from becoming a ecological disaster."`
+			choice
+				`	"This is scary stuff. Is there any way I can help?"`
+                                `       "Wow. What can we do?"`
+
+                       `        "There is somthing you can do. We need some cargo ships escorted to Alphard to pick up some super-strong nanotube buckyballs that hopefully should help strengthen this." He gestures toward the tower. "I also need you to bring some scientists back as well. Are you up to the challange?"`
```
Comment (jafdy) [inline data/transport missions.txt]: Nanotubes and Buckyballs are different shapes so you can't have both.

**Comment by Gkramme** [comment]
Thanks, guys! I really appreciate the help you're giving me :)

**Comment by jafdy** [comment]
It comes with practice. Also can I recommend making a new branch for each pull, you can do several thing at once.

**Comment by Gkramme** [comment]
This is what it looks like when I edit it:
image

File: data/transport missions.txt [hunk 12]
```diff
@@ -1246,3 +1246,110 @@ mission "Terraforming 5"
...
+	npc
+		personality nemesis waiting
+		government Pirate
+		fleet "Small Northern Pirates"
+        npc
+		personality nemesis waiting
+		government Pirate
+		fleet "Large Northern Pirates"
+	npc
+		personality nemesis waiting
```
Comment (jafdy) [inline data/transport missions.txt]: Your Pirate fleets are in orbit around Earth at the moment. If you set them to `waiting` in systems between Earth and Alphard or around Alphard. They will actively seek out the player if they are not `staying`. Also as you get a navy license at the end of this I would recommend making it hard.
Last thing, you don't have any ships or fleets for the return trip. And Behemoths are quite tough so they may be quite easy to defend. If they are being use to build a large tower then you may want more of them.

**Comment by jafdy** [comment]
You can update the master by creating a new pull request from this repo, click compare across forks, change the base to your master, make the pull and merge.

**Comment by Gkramme** [comment]
@endless-sky is there any more issues with this?

**Comment by endless-sky** [comment]
I really don't think you should be able to get a Navy license that easily. Also, there's still a bunch of typos.

Is "skytower" a term that is used elsewhere? I've never heard that term used for a space elevator, so it might be confusing to people.

**Comment by jafdy** [comment]
I think that it is a very large sky scraper. And yes the Navy license needs to be harder to get. At the very lest make it a dozen bulk freighter and too many pirates.

**Comment by Gkramme** [comment]
@endless-sky skytower flows a bit smoother than space elevator, but if the latter clears up confusion, I will change it.
@jafdy I'll change it.

**Comment by Gkramme** [comment]
@endless-sky where are the typos?

File: data/transport missions.txt [hunk 13]
```diff
@@ -1246,3 +1246,195 @@ mission "Terraforming 5"
...
+	destination Earth
+	to offer 
+               has "building skytower 1: done"
+	passengers 4
+       
+        on complete
+        set licence navy
+	
+	on offer
+		diolouge
```
Comment (jafdy) [inline data/transport missions.txt]: dialog

File: data/transport missions.txt [hunk 14]
```diff
@@ -1246,3 +1246,195 @@ mission "Terraforming 5"
...
+		fleet "Large Northern Pirates"
+	
+        npc accompany save
+		government Republic
+		personality escort timid
+		fleet
+			name "RNS Aber"
+			variant
+				"Behemoth"
+        T accompany save
```
Comment (jafdy) [inline data/transport missions.txt]: npc accompany save

**Comment by Gkramme** [comment]
I deleted my fork, so if any changes or fixes are to be made, @endless-sky has to merge it first

**Comment by Amazinite** [comment]
Given the fact that it's super easy to get ships off of a planet using repulsor engines (see 2360 in the historical timeline page of the wiki), wouldn't it be highly inefficient to build a space elevator for the purpose of getting people off planet? It would probably make more sense to use a few fleets of transport ships to ferry people off world. Expensive space elevators that have the risk of possibly falling as explained in the mission just don't seem practical, and is much more something that a single planet species might make. The main point behind a space elevator is to get stuff into space while requiring less energy, but that's already done by the repulsor engines. 

Another thing is the wording when the guy is explaining why they're making he space elevator. To my understanding, Earth is already an almost entirely urban planet (see Regions of Space wiki and Earth's planet description), and given the current predictions of the ice caps melting in only a few decades or so, it makes sense that this future Earth that has been overpopulated and overpolluted for years would already be long passed the point of the ice caps melting. 

It's a cool idea, i just don't think it makes sense. :/

**Comment by Gkramme** [comment]
A space elevator would be able to carry up lots of people for a cheap price. It would be a milestone for technological acheivment. It also would be cool :)

**Comment by Amazinite** [comment]
But like i said, would it not be easier to use transport ships and maybe even bulk freighters with their repulsor engines to do it same thing? Space elevators just seem like they would only be practical if the repulsor engines didn't exist, and are really only meant for low tech civilizations with no easy way to get to space.

**Comment by jafdy** [comment]
If you want a cool factor have it on one of the paradise planets. They like to show off.

**Comment by Gkramme** [comment]
Earth needs it the most tho.

**Comment by Amazinite** [comment]
But it doesn't really need it at all at the same time. If a space elevator was better and cheaper at getting stuff off a planet, then the repulsor engines wouldn't be used and planets would likely all have a space elevator. One disaster over eight hundred years ago would likely have only caused people to try and advance the technology further instead of out right banning them.

**Comment by Gkramme** [comment]
Well, would you spend half a trillion dollars to show off?

**Comment by Amazinite** [comment]
The paradise worlds would.

**Comment by Gkramme** [comment]
Earth needs it the most. The Republic would build one on Earth to get rid of people and strengthen their position of power.

**Comment by Amazinite** [comment]
Repulsor engines.

**Comment by Gkramme** [comment]
Just for lore reasons

**Comment by Amazinite** [comment]
If you're going to use this idea, then you should really address why they can't just use transport ships. Maybe the elevator is so huge that it would be able to transport more people than a constant stream of transport ships, or the shipyards are too crowded to accommodate enough transport ships to move a reasonable number of people, or to reserve enough landing pads to use all those transport ships would require postponing freighters from landing that provide the food to keep the planet's population alive.

**Comment by Amazinite** [comment]
Again, I like the premise, it's just that it doesn't seem logical to me, at least not in the way that I have interpreted the lore.

**Comment by Gkramme** [comment]
Again, I can't really edit this because I deleted the branch, so if someone could find a way to, that would be helpful.

**Comment by Amazinite** [comment]
You could just go to "files changed" and copy what is there.

**Comment by Gkramme** [comment]
Of course. I was actually hoping someone would do that.

**Comment by Amazinite** [comment]
Reworked the relocation jobs.
[Relocation Jobs.txt](https://github.com/endless-sky/endless-sky/files/287559/Relocation.Jobs.txt)

Removed all the errors from the skytower/space elevator job, but I have yet to go through and change the text to be more lore friendly.

**Comment by Gkramme** [comment]
Nice! Looks good. Maybe you could start a new pr?

**Comment by Amazinite** [comment]
I want to wait till I have the space elevator mission done first then pr them together, since they're technically related. Plus, there's plenty of time till v0.9.2 now.

**Comment by Amazinite** [comment]
Remember when I talked about possibly having a different earth sprite after the space elevator is built?

[planet.zip](https://github.com/endless-sky/endless-sky/files/294218/planet.zip)

put these in a plugin folder or something and switch earth's sprite to "earthse"

**Comment by Amazinite** [comment]
Done.

**Comment by Gkramme** [comment]
Alright.

**Comment by jafdy** [comment]
@Gkramme you should close this now.

**Comment by Gkramme** [comment]
Done

**Pull request closed without merging**